extensions/wrapper/wrapper/src/test/java/de/sovity/edc/ext/wrapper/api/offering/OfferingResourceTest.java

```diff
@@ -0,0 +1,115 @@
+package de.sovity.edc.ext.wrapper.api.offering;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.restassured.http.ContentType;
+import io.restassured.response.ValidatableResponse;
+import org.eclipse.edc.junit.annotations.ApiTest;
+import org.eclipse.edc.junit.extensions.EdcExtension;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.io.File;
+import java.io.IOException;
+
+import static de.sovity.edc.ext.wrapper.TestUtils.createConfiguration;
+import static de.sovity.edc.ext.wrapper.TestUtils.givenManagementEndpoint;
+
+@ApiTest
+@ExtendWith(EdcExtension.class)
+class OfferingResourceTest {
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private static JsonNode contractOfferValid;
+    private ObjectNode contractOffer;
+
+    private static final String ASSET_ENTRY_KEY = "assetEntryDto";
+    private static final String POLICY_DEFINITION_REQUEST = "policyDefinitionRequestDto";
+    private static final String CONTRACT_DEFINITION_REQUEST = "contractDefinitionRequestDto";
+    private static final String MISSING_PROPERTY_TEXT = "missing required creator property";
+    private static final String NO_REQUEST_BODY_ERROR = "No CreateOfferingDto provided";
+
+    @BeforeAll
+    static void init() throws IOException {
+        ClassLoader classLoader = OfferingResourceTest.class.getClassLoader();
+        contractOfferValid = MAPPER.readTree(new File(
+                classLoader.getResource("offering/contract-offer-valid.json").getFile()));
+    }
+
+    @BeforeEach
+    void setUp(EdcExtension extension) throws JsonProcessingException {
+        extension.setConfiguration(createConfiguration());
+        contractOffer = (ObjectNode) MAPPER.readTree(contractOfferValid.toString());
+    }
+
+    ValidatableResponse whenCreateOfferingEndpoint(String body) {
+        return givenManagementEndpoint()
+                .when()
+                .contentType(ContentType.JSON)
+                .body(body)
+                .post("/wrapper/offering/contract-offer")
+                .then()
+                .log()
+                .all();
+    }
+
+    ValidatableResponse whenJsonPropertyMissing(String property) {
+        contractOffer.remove(property);
+        return whenCreateOfferingEndpoint(contractOffer.toString());
+    }
+
+    @Test
+    void shouldCreateOffer() {
+        whenCreateOfferingEndpoint(contractOfferValid.toString())
+                .assertThat()
+                .statusCode(204);
```

Comment:
FYI generated clients mostly only care about 200<=statusCode<300 and the exception class